src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -367,9 +367,11 @@ private static bool IsExpressionOfArgumentInDeconstruction(ExpressionSyntax expr
                 expression == memberAccess.Expression)
             {
                 var symbol = semanticModel.GetSymbolInfo(memberAccess, cancellationToken).Symbol;
-                if (symbol is IMethodSymbol { MethodKind: MethodKind.ReducedExtension, ReducedFrom: IMethodSymbol reducedFrom } &&
-                    reducedFrom.Parameters.Length > 0 &&
-                    reducedFrom.Parameters.First().RefKind == RefKind.Ref)
+                if (symbol is IMethodSymbol
+                    {
+                        MethodKind: MethodKind.ReducedExtension,
+                        ReducedFrom.Parameters: [{ RefKind: RefKind.Ref }],
```

Comment:
To match previous logic this should be
```suggestion
                        ReducedFrom.Parameters: [{ RefKind: RefKind.Ref }, ..],
```

Reply:
ick.  good catch.  going to audit this now.  clearly some of my logic with lengths isn't right.